merlin/models/tf/metrics/evaluation.py

```diff
@@ -0,0 +1,292 @@
+        tf.assert_equal(
+            tf.rank(predictions), 2, f"predictions must be 2-D tensor (got {predictions.shape})"
+        )
+        if labels is not None:
+            tf.assert_equal(tf.rank(labels), 2, f"labels must be 2-D tensor (got {labels.shape})")
+            labels = tf.cast(labels, tf.int32)
+        return labels, tf.cast(predictions, tf.int32)
+
+    def get_config(self):
+        config = {}
+        if type(self) is PopularityMetric:
+            # Only include function argument when the object is of a subclass.
+            config["fn"] = self._fn
+            config["k"] = self.k
+            config["is_prob_distribution"] = self._fn_kwargs.get("is_prob_distribution", True)
+
+            for k, v in self._fn_kwargs.items():
+                config[k] = backend.eval(v) if is_tensor_or_variable(v) else v
+            base_config = super(PopularityMetric, self).get_config()
+            return dict(list(base_config.items()) + list(config.items()))
+
+    @classmethod
+    def from_config(cls, config):
+        fn = config.pop("fn", None)
+        k = config.pop("k", None)
+        is_prob_distribution = config.pop("is_prob_distribution", None)
+        if cls is PopularityMetric:
+            return cls(get_metric(fn), k=k, is_prob_distribution=is_prob_distribution, **config)
+        return super(PopularityMetric, cls).from_config(config)
+
+
+@metrics_registry.register_with_multiple_names("novelty_at", "novelty")
+class NoveltyAt(PopularityMetric):
+    def __init__(self, item_freq_probs, k=10, is_prob_distribution=False, name="novelty"):
+        super().__init__(
+            novelty_at, item_freq_probs, k=k, is_prob_distribution=is_prob_distribution, name=name
+        )
+
+
+@metrics_registry.register_with_multiple_names("popularity_bias_at", "popularity_bias")
+class PopularityBiasAt(PopularityMetric):
+    def __init__(self, item_freq_probs, k=10, is_prob_distribution=False, name="novelty"):
+        super().__init__(
+            popularity_bias_at,
+            item_freq_probs,
+            k=k,
+            is_prob_distribution=is_prob_distribution,
+            name=name,
+        )
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class ItemCoverageAt(tf.keras.metrics.Metric):
+    """Computes the prediction coverage for a top-k recommendation model.
+
+    Parameters
+    ----------
+    num_unique_items : int
+        The number of unique items in the train set.
+    k : int, optional
+        The cut-off for coverage metric, by default 5
+
+    Returns
+    ----------
+    coverage: float
+        The prediction coverage of the recommendations at the k-th position.
+    """
+
+    def __init__(self, num_unique_items: int, k=5, name: str = None, dtype=None, **kwargs):
+        if name is not None:
+            name = f"{name}_{k}"
+        super().__init__(name=name, dtype=dtype)
+
+        self.k = k
+        self.num_unique_items = num_unique_items
```

Comment:
Maybe we could make `self.num_unique_items` a scalar non-trainable variable, and have an `update()` method to allow for users to update `num_unique_items` (with variable assign), as that might be useful when training models with streaming data, as the (number of )available items at a given point of time might change depending on the domain (e.g. on news recommendation)